Automattic/wp-calypso
# Hosting onboarding: Allow dev accounts to be set when using social signup.

Closes https://github.com/Automattic/dotcom-forge/issues/2529

## Proposed Changes

* Adds `is_dev_account` to social signup properties to flag a user as a developer when they sign up using a social account.

## Testing Instructions


In order for social accounts to work we have to setup local environment with production build.
- Run `bin/calypso-secrets .config/secrets.php` in your WordPress.com sandbox;
- Copy the output of the script to `config/secrets.json` in your wp-calypso checkout;
- Build this branch with `NODE_ENV=production CALYPSO_ENV=production BUILD_TRANSLATION_CHUNKS=true yarn run build`;
- Run these commands in the Calypso root folder so you can SSL the local version you just built:

```sh
openssl req \
    -newkey rsa:2048 \
    -x509 \
    -nodes \
    -keyout ./config/server/key.pem \
    -new \
    -out ./config/server/certificate.pem \
    -subj /CN=wordpress.com \
    -reqexts SAN \
    -extensions SAN \
    -config <(cat /System/Library/OpenSSL/openssl.cnf \
     <(printf '[SAN]\nsubjectAltName=DNS:wpcalypso.wordpress.com')) \
    -sha256 \
    -days 365
 
# Add it to the trusted certificates
sudo security add-trusted-cert -d -k $(security list-keychains | grep System | cut -d '"' -f 2 ) ./config/server/certificate.pem
```

- Kick off the server with `NODE_ENV=production HOST=wpcalypso.wordpress.com PORT=443 PROTOCOL=https CALYPSO_ENV=wpcalypso BUILD_TRANSLATION_CHUNKS=true node build/server.js`;
- Proxy `wpcalypso.wordpress.com` to `127.0.0.1`;
- go to `/start/hosting?flags=hosting-onboarding-i2`
- Signup with either google or apple with `dev` checkbox selected
- In network tab verify that a call to `/users/social/new` with `is_dev_account:true` in the request payload

Test end to end
- Apply patch D112495-code to sandbox
- Verify account created with social is flagged as developer

## Pre-merge Checklist
